src/components/FormInput/FormInput.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import { DEFAULT_PROPS } from './FormInput.constants';
+import { StyledFormInput } from './FormInput.styles';
+
+import type { FormInputProps } from './FormInput.types';
+
+export const FormInput: React.FC<FormInputProps> = ({
+  variant = DEFAULT_PROPS.variant,
+  type = DEFAULT_PROPS.type,
+  size = DEFAULT_PROPS.size,
+  placeholder,
+  value,
+  onChange,
+  errorMessage,
+  disabled = DEFAULT_PROPS.disabled,
+  onFocus,
+  onBlur,
+  'data-testid': dataTestId,
+  ...rest
+}) => {
+  const isError = !!errorMessage;
+
+  const handleFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (onFocus) onFocus(event);
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (onBlur) onBlur(event);
+  };
+
+  return (
+    <StyledFormInput
```

Comment:
The component passes props to StyledFormInput using standard prop names (variant, size, error) while the styled component expects $variant, $size, $error (see styles). This mismatch prevents the styled layer from receiving the intended values and may lead to styling not being applied.

**Sugerencia:**
```typescript
Pass the $-prefixed props to the styled component in addition to the original props if needed by MUI. Example:

<StyledFormInput
  $variant={variant}
  $size={size}
  $error={isError}
  variant={variant}
  size={size}
  error={isError}
  ...
/>

This keeps MUI props intact and supplies the styled component with the values it expects.
```